woodwork/statistics_utils/frequency_inference/_generate_freq_candidates.py

```diff
@@ -0,0 +1,43 @@
+import pandas as pd
+from ._constants import NON_INFERABLE_FREQ, WINDOW_LENGTH, CANDIDATE_COLUMN_NAME
+
+
+def _generate_freq_candidates(time_series):
+    """Calculate a set of candidate frequecies for incoming time_series
+
+    Args:
+        time_series (Series): The time_series for which candidate frequencies will be calculated.
+
+    Returns:
+        (dict[pandas_alias(str) -> dict]): Returns a dictionary where each key is candidate Pandas Frequency Alias and the value is
+        a dictionary containing the following keys:
+            - alias (str): The Pandas Freq Alias
+            - count (int): The number of windows where this alias is valid
+            - min_dt (pd.TimeStamp): The earliest timestamp for this frequency.
+            - max_dt (pd.TimeStamp): The latest timestamp for this frequency.
+    """
+    
+    alias_dict = {}
+    for window in time_series.rolling(WINDOW_LENGTH):
+        if len(window) == WINDOW_LENGTH:
```

Comment:
If the series has fewer rows than the window length, you don't get any frequency calculations other than the frequency of the whole series. Is this okay for what we want? If so, we should note it somewhere. If not, then we might as well make the minimum length the length of the window instead of 3